vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNavItem.java

```diff
@@ -280,11 +301,58 @@ public String getPath() {
      */
     public void setQueryParameters(QueryParameters queryParameters) {
         this.queryParameters = queryParameters;
-        refresh();
+        // Apply new query parameters to the path
+        setPath(getPath());
     }
 
-    private void refresh() {
-        setPath(getPath());
+    /**
+     * Gets the path aliases for this item.
+     *
+     * @return the path aliases for this item, empty if none
+     */
+    public Set<String> getPathAliases() {
+        Serializable pathAliasesRaw = getElement()
```

Comment:
Would it work if we cast the property to JsonArray right here?

```suggestion
        JsonArray pathAliasesRaw = (JsonArray) getElement()
```